powerscale/provider/smb_share_settings_resource.go

```diff
@@ -0,0 +1,565 @@
+	if !ok {
+		res.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *c.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+	r.client = pscaleClient
+}
+
+// Create allocates the resource.
+func (r *SmbShareSettingsResource) Create(ctx context.Context, request resource.CreateRequest, response *resource.CreateResponse) {
+	tflog.Info(ctx, "creating smb share settings")
+
+	var sharePlan models.SmbShareSettingsResourceModel
+	diags := request.Plan.Get(ctx, &sharePlan)
+	//cachedPermission := sharePlan.Permissions
+
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	zone := sharePlan.Zone.ValueString()
+
+	var toUpdate powerscale.V7SmbSettingsShareSettings
+	// Get param from tf input
+	err := helper.ReadFromState(ctx, &sharePlan, &toUpdate)
+	if err != nil {
+		errStr := constants.ReadSMBShareSettingsErrorMsg + "with error: "
+		message := helper.GetErrorString(err, errStr)
+		response.Diagnostics.AddError("Error creating smb share settings",
+			fmt.Sprintf("Could not read smb share settings param with error: %s", message),
+		)
+		return
+	}
+	err = helper.UpdateSmbShareSettings(ctx, r.client, toUpdate, zone)
+	if err != nil {
+		errStr := constants.UpdateSMBShareSettingsErrorMsg + "with error: "
+		message := helper.GetErrorString(err, errStr)
+		response.Diagnostics.AddError("Error creating smb share settings",
+			message)
+		return
+	}
+	/*tflog.Debug(ctx, fmt.Sprintf("smb share  %s created", shareID.Id), map[string]interface{}{
+		"smbShareResponse": shareID,
+	})
+	*/
+	getShareSettingsResponse, err := helper.GetSmbShareSettings(ctx, r.client, zone)
+	if err != nil {
+		errStr := constants.ReadSMBShareSettingsErrorMsg + "with error: "
+		message := helper.GetErrorString(err, errStr)
+		response.Diagnostics.AddError("Error creating smb share settings",
+			message)
+		return
+	}
+
+	var state models.SmbShareSettingsResourceModel
+	err = helper.CopyFieldsToNonNestedModel(ctx, getShareSettingsResponse.Settings, &state)
+	if err != nil {
+		response.Diagnostics.AddError(
+			"Error copying fields of smb share settings resource", err.Error(),
+		)
+		return
+	}
+	state.Zone = sharePlan.Zone
+	state.ID = sharePlan.Zone
+	diags = response.State.Set(ctx, state)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+	tflog.Info(ctx, "create smb share settings completed")
+}
+
+// Read reads the resource state.
+func (r SmbShareSettingsResource) Read(ctx context.Context, request resource.ReadRequest, response *resource.ReadResponse) {
+	tflog.Info(ctx, "reading smb share settings")
+	var shareState models.SmbShareSettingsResourceModel
+	diags := request.State.Get(ctx, &shareState)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	zone := shareState.Zone.ValueString()
+	//scope := shareState.Scope.ValueString()
```

Comment:
also, if this line is useless, please remove it
and the resource has no scope param?